From da990536eca09c6de74627541cd56ecfad925eda Mon Sep 17 00:00:00 2001
From: Chris Wilson <chris@chris-wilson.co.uk>
Date: Thu, 03 Feb 2011 09:41:48 +0000
Subject: uxa: Undo damage translation before appending

The region is used to paint onto the backing pixmap (and thus
translated) prior to being passed to the damage layer (wrt to the
drawable). So the local translation needs to be undone first.

Identified by Christopher James Halse Rogers.

Bugzilla: https://bugs.freedesktop.org/show_bug.cgi?id=33650
Signed-off-by: Chris Wilson <chris@chris-wilson.co.uk>
---
diff --git a/uxa/uxa-render.c b/uxa/uxa-render.c
index cf8b8ea..02bfa03 100644
--- a/uxa/uxa-render.c
+++ b/uxa/uxa-render.c
@@ -1090,6 +1090,7 @@ try_solid:
 	/* XXX xserver-1.8: CompositeRects is not tracked by Damage, so we must
 	 * manually append the damaged regions ourselves.
 	 */
+	pixman_region_translate(&region, -dst_x, -dst_y);
 	DamageRegionAppend(dst->pDrawable, &region);
 
 	pixman_region_fini(&region);
--
cgit v0.8.3-6-g21f6
